File: LEETCODE/LEETCODE-CPP/153.cpp
// 153. Find Minimum in Rotated Sorted Array
// https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/description/

class Solution
{
public:
    int findMin(vector<int> &nums)
    {
        int n = nums.size();
        int start = 0, end = n - 1;
        int ans = INT_MAX;

        while (start < end)
        {
            int mid = start + (end - start) / 2;

            if (nums[mid] > nums[end])
            {
                start = mid + 1;
            }
            else
            {
                end = mid;
            }
        }

        return nums[start];
    }
};

File: LEETCODE/LEETCODE-PYTHON/153.py
# 153. Find Minimum in Rotated Sorted Array
# https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/

# Method-1 using binary search (TC-O(logn))
class Solution:
    def findMin(self, nums: List[int]) -> int:
        low,high= 0, len(nums)-1

        while low<high:
            mid=low+(high-low)//2

            if nums[mid]>nums[high]:
                low=mid+1
            else:
                high=mid

        return nums[low]